src/box/lua/net_replicaset.lua

```diff
@@ -0,0 +1,376 @@
+local fiber = require('fiber')
+local net_box = require('net.box')
+
+-- The constant is copy-paste from tarantool/src/lua/socket.lua.
+local TIMEOUT_INFINITY = 500 * 365 * 86400
+
+-- Error type used for all errors raised in net.replicaset.
+local ERROR_TYPE = 'NetReplicasetError'
+
+-- Map of errors by their codes.
+local error_list = {
+    [1] = {
+        name = 'REPLICASET_NOT_FOUND',
+        message = 'The replicaset was not found by its name',
+    },
+    [2] = {
+        name = 'NO_WRITABLE',
+        message = 'Writable instance was not found in replicaset',
+    },
+    [3] = {
+        name = 'MORE_THAN_ONE_WRITABLE',
+        message = 'More than one writable was found in replicaset',
+    },
+}
+
+-- Map of error codes by error name.
+local error_codes = {}
+for code, err in ipairs(error_list) do
+    error_codes[err.name] = code
+end
+
+-- Raise a net.replicaset specific error.
+local function rs_error(err_code, replicaset_name, payload)
+    local error_info = error_list[err_code]
+    assert(error_info)
+    local args = payload and table.copy(payload) or {}
+    args.type = ERROR_TYPE
+    args.code = err_code
+    args.name = error_info.name
+    args.message = error_info.message
+    args.replicaset = replicaset_name
+    box.error(args)
```

Comment:
What's the point in introducing the new type of errors. Why not reuse the existing type `ClientError` used for all other Tarantool errors? We can add the new error codes to `src/box/errcode.h`.

Reply:
Yes, I think that it was my mistake (to use dedicated error type) if the module is a Core module, it should use Core error type.

Reply:
Remade using ClientError.